app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/transferdetails/TransferDetailsViewModel.kt

```diff
@@ -69,6 +72,34 @@ class TransferDetailsViewModel @Inject constructor(
 
     fun getTransferUrl(transferUuid: String): String = sharedApiUrlCreator.shareTransferUrl(transferUuid)
 
+    fun startDownloadingAllFiles(transfer: TransferUi) {
+        viewModelScope.launch {
+            val url = sharedApiUrlCreator.downloadFilesUrl(transfer.uuid) ?: return@launch
+            val userAgent = buildUserAgent(
+                appId = BuildConfig.APPLICATION_ID,
+                appVersionCode = BuildConfig.VERSION_CODE,
+                appVersionName = BuildConfig.VERSION_NAME,
+            )
+            when (transfer.files.size) {
+                1 -> {
+                    val singleFile = transfer.files.single()
+                    DownloadManagerUtils.scheduleDownload(
+                        url = sharedApiUrlCreator.downloadFileUrl(transfer.uuid, singleFile.uid) ?: return@launch,
```

Comment:
Maybe we shouldn't fail silently here, and show a snackbar to the user that it didn't work

Reply:
it'll be change In an upcoming PR that depends on a new KMP release